lint: add lint rule for foribidding redact.Unsafe() method.

aa-joshi opened the pull request:
Previously, we introduced `encoding.Unsafe` wrapper on `redact.Unsafe` as part of #133144. This patch adds linting rule for `redact.Unsafe` method so that future work can properly infer whether objects which are being logged are safe or unsafe.

Epic: CRDB-37533
Part of: CRDB-42324
Release note: None

snapshot for validation:
<img width="966" alt="Screenshot 2024-11-13 at 5 30 48 PM" src="https://github.com/user-attachments/assets/09767e7e-d797-4c53-b568-c0c910286176">

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/135140)
<!-- Reviewable:end -->

arjunmahishi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/135140#-:-OCCooWZ6eEtsybglrVL:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

aa-joshi committed:
lint: add lint rule for foribidding redact.Unsafe() method.
Previously, we introduced `encoding.Unsafe` wrapper on `redact.Unsafe` as part
of cockroachdb #133144. This patch adds linting rule for `redact.Unsafe`
method so that future work can properly infer whether objects which are being
logged are safe or unsafe.

Epic: CRDB-37533
Part of: CRDB-42324
Release note: None

aa-joshi commented:
TFTR!

bors r+
